### PR TITLE
Add Privacy Policy and TOS

### PR DESCRIPTION
**Pull request by hanars**
This adds a terms of service and privacy policy to seqr and forces users to accept them. It also adds a cookie use policy to seqr and forces them to accept that as well. 

For the privacy policy/ TOS, a key requirement is that it is easy to force users to re-accept policies when they change. Therefore, we store the version of the policies that users accepted, and whenever we need to update our policies we update the version and seqr will detect that everyone is out of date, without needing to run any sort of database migration. 

For the cookie policy, we do't have any reporting/auditing obligations the way we do with the privacy and TOS, so we just store that users have accepted using cookies in the cookies rather than the database. This means every time a user logs in in a different browser they **will not** need to reaccept TOS and privacy, but **will** need to reaccept cookies, which seems fine to me

### CODE REVIEW

File: ui/pages/Public/PrivacyPolicy.jsx [hunk 1]
```diff
@@ -0,0 +1,402 @@
+import React from 'react'
+import { Header, Segment, List } from 'semantic-ui-react'
+
+export default () =>
+  <Segment basic padded="very">
```
Comment (hanars) [inline ui/pages/Public/PrivacyPolicy.jsx:5]: The text for this policy comes from nd is approved by Broad legal. Theres no real need to review it

File: ui/pages/Public/TermsOfService.jsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
+import React from 'react'
+import { Header, Segment, List } from 'semantic-ui-react'
+
+export default () =>
+  <Segment basic padded="very">
```
Comment (hanars) [inline ui/pages/Public/TermsOfService.jsx:5]: The text for this policy comes from and is approved by Broad legal. Theres no real need to review it

**Comment by hanars** [comment]
passing test: https://travis-ci.org/github/broadinstitute/seqr/builds/740908450

**Comment by hanars** [comment]
passing codacy: https://app.codacy.com/gh/macarthur-lab/seqr/pullRequest?prid=6441558

**Review by ShifaSZ — CHANGES_REQUESTED** [review]

**Comment by ShifaSZ** [comment]
There is something like a string `ue` at the top left of the dashboard page. I'm not sure what it is.
image

File: ui/shared/components/page/AcceptPolicies.jsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      // eslint-disable-next-line jsx-a11y/label-has-for
+      <label>
+        I accept the <Link target="_blank" to="/terms_of_service">Terms of Service</Link> and
+        the <Link target="_blank" to="/privacy_policy">Privacy Policy</Link>
+      </label>
+    ),
+  },
+]
+
+export const AcceptPolicies = React.memo(({ user, onSubmit }) => (
```
Comment (ShifaSZ) [inline ui/shared/components/page/AcceptPolicies.jsx:57]: There should not be an `export` here.

**Review by ShifaSZ — CHANGES_REQUESTED** [review]

File: seqr/views/apis/users_api_tests.py [hunk 4]
```diff
@@ -261,3 +262,34 @@ def test_forgot_password(self, mock_send_mail):
...
+        self.assertDictEqual(response.json(), {'currentPolicies': True})
+
+        new_policy = UserPolicy.objects.get(user=self.no_access_user)
+        self.assertEqual(new_policy.privacy_version, 1.0)
+        self.assertEqual(new_policy.tos_version, 1.0)
+
+        # Test updating user with out of date policies
+        existing_policy = UserPolicy.objects.get(user=self.manager_user)
+        self.assertNotEqual(existing_policy.privacy_version, 1.0)
+        self.assertNotEqual(existing_policy.tos_version, 1.0)
```
[the remark this answers is not in the document]
Comment (hanars) [reply seqr/views/apis/users_api_tests.py:286]: No, the version number is never None. Its a foreign one-to-one database relationship, so if the relationship is not set there is no model (see line 267 where we assert that the model does not exist at all). This fixture data creates a relationship with outdated versions, so we are confirming that before the update in the database there are versions set but they are not correct, and after updating they are the correct versions

File: seqr/views/react_app.py [hunk 5]
```diff
@@ -6,29 +6,40 @@
...
 
 
+def _get_json_for_initial_user(user):
+    user_json = _get_json_for_user(user)
+
+    user_json['currentPolicies'] = False
+    if hasattr(user, 'userpolicy'):
+        current_privacy = user.userpolicy.privacy_version
+        current_tos = user.userpolicy.tos_version
+        user_json['currentPolicies'] = current_privacy == SEQR_PRIVACY_VERSION and current_tos == SEQR_TOS_VERSION
```
[the remark this answers is not in the document]
Comment (hanars) [reply seqr/views/react_app.py:39]: that makes a really long line so then we have to do line breaking and I don't think that is any more readable?

File: seqr/views/react_app_tests.py [hunk 6]
```diff
@@ -8,6 +8,8 @@
 
 INITIAL_JSON_REGEX = r'\(window\.initialJSON=(?P<initial_json>[^)]+)'
 
+MAIN_APP_USER_FIELDS = {'currentPolicies'}
+MAIN_APP_USER_FIELDS.update(USER_FIELDS)
```
[the remark this answers is not in the document]
Comment (hanars) [reply seqr/views/react_app_tests.py:12]: No, `update` does not return anything

File: ui/shared/components/page/BaseLayout.jsx [hunk 7]
```diff
@@ -8,6 +8,7 @@ import PropTypes from 'prop-types'
 import styled from 'styled-components'
 import { Grid } from 'semantic-ui-react'
 
+import AcceptPolicies from './AcceptPolicies'
```
[the remark this answers is not in the document]
Comment (hanars) [reply ui/shared/components/page/BaseLayout.jsx:11]: not at all, there is the non-connected base class that is exported for testing purposes only which would be accessed with `import { AcceptPolicies } from './AcceptPolicies'`, and then there is the connected, routed default export which is what we want here and what we are correctly importing. This style is used throughout our repo

**Comment by hanars** [comment]
> There is something like a string `ue` at the top left of the dashboard page. I'm not sure what it is.
> image

I don't have that locally 
Screen Shot 2020-11-03 at 11 56 09 AM

File: ui/shared/components/page/AcceptPolicies.jsx (hunk 3 above)
Comment (hanars) [reply ui/shared/components/page/AcceptPolicies.jsx:57]: this is needed for testing, I will change this to `BaseAcceptPolicies` and add a comment to make it clearer

File: seqr/views/react_app_tests.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (ShifaSZ) [reply seqr/views/react_app_tests.py:12]: I see.

File: ui/shared/components/page/BaseLayout.jsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (ShifaSZ) [reply ui/shared/components/page/BaseLayout.jsx:11]: Maybe, it is less confusing to use a different name for the default exported function than the name exported.

File: ui/shared/components/page/AcceptPolicies.jsx [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+  primary: true,
+  content: 'Accept',
+  onClick: () => {
+    Cookies.set('accepted_cookies', true)
+    window.location.href = window.location.href
+  },
+}]
+
+const AcceptCookies = () =>
+  Cookies.get('accepted_cookies') ||
```
Comment (ShifaSZ) [inline ui/shared/components/page/AcceptPolicies.jsx:24]: It causes an un-wanted string `true` on the top left of the screen as shown in the below picture.
image
If change it to `!Cookies.get('accepted_cookies') &&` then the string `true` will disappear.

**Review by ShifaSZ — CHANGES_REQUESTED** [review]
The issue of an extra string `true` on the top left of the screen needs to be fixed.

**Review by ShifaSZ — APPROVED** [review]

**Comment by hanars** [comment]
Passing tests: https://travis-ci.org/github/broadinstitute/seqr/builds/741670236

**Comment by hanars** [comment]
passing codacy: https://app.codacy.com/gh/macarthur-lab/seqr/pullRequest?prid=6441558

**Pull request merged**